docs/design/proposals/kms-custom-http.md

```diff
@@ -0,0 +1,40 @@
+# External Storage of Encryption Passphrases
+
+## Goal
+
+The current KMS methods are modeled to provide an end-to-end solution for managing the encryption passphrases that are used by ceph-csi. While they are fine and use top-notch services like Vault and Amazon KMS, they do struggle to fit into configurations where the encryption is handleded completely by a dedicated in-house service.
```

Comment:
Can there be a simple open source service provided that ceph-csi can use for automated e2e testing? A container-image that the e2e tests can deploy would be ideal.

Reply:
A good suggestion, but the differentiator here is to completely customize the API and not align on predefined schemes. for predefined APIs for this we already supports aws, vault, ...

Reply:
The KMS implementation uses defined endpoints as mentioned at the bottom of the document. A very simple KMS service would help to validate the implementation and functioning of Ceph-CSI. The KMS service could run completely in-memory, with a simple `map[string]string` kind of structure that handles the endpoints.

Reply:
sure, for validation it would be very convenient